roles/vault/tasks/shared/sync_file.yml

```diff
@@ -28,40 +27,37 @@
   stat:
     path: "{{ sync_file_key_path }}"
   register: sync_file_key_stat
-  when: sync_file_is_cert|d()
 
 - name: "sync_file | Combine all possible file sync sources"
   set_fact:
     sync_file_srcs: "{{ sync_file_srcs|default([]) + [host_item] }}"
-  with_items: "{{ sync_file_hosts | unique }}"
+  with_items: "{{ sync_file_hosts|default() | unique }}"
```

Comment:
This should be:
```
  with_items: "{{ sync_file_hosts| default([]) | unique }}"
```
Otherwise it will return an object `set([])`, which might end up being an issue iterating over later.

Reply:
Nevermind, this is just an iterator itself. This will not propagate. Just keep it as is